File: B_Index_and_Maximum_Value.java
import java.util.*;
public class B_Index_and_Maximum_Value {
    public static void main(String[] args) {
        Scanner sc =  new Scanner(System.in);
        int t = sc.nextInt();
        while(t > 0){
           
            int n = sc.nextInt();
            int m = sc.nextInt();
           int[]arr = new int[n];
           
            for(int i = 0; i < n; i++){
                arr[i] = sc.nextInt();
            }
            //array is created
    
            for(int i = 0; i < m;i++){
                
                char s = sc.next().charAt(0); 
              
                int l = sc.nextInt();
                int r = sc.nextInt();
                if(s == '+'){
                  
                    
                  for(int j = 0; j < n;j++){
                     if(l <=arr[j] && r>=arr[j]){
                            arr[j]++;
                        }
                  }
                }
                else{
                     
                    
                    for(int j = 0; j < n;j++){
                     if(l <=arr[j] && r>=arr[j]){
                            arr[j]--;
                        }
                    }
                }
                int max = Integer.MIN_VALUE;
                for(int k = 0; k < n; k++){
                    if(arr[k] > max){
                        max = arr[k];
                    }
                }
                System.out.print(max + " ");
                
               
               
              
            }    
             System.out.println();
            t--;
           
        } // main loop
    }
}

File: B_Index_and_Maximum_Value.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int t;cin >> t;
    while(t--){
        int n,m;cin>>n>>m;
        vector<int> arr(n);
       
        for(int i = 0; i < n; i++){
            cin >> arr[i];
        }
        //array is created

        vector<int> diff(n+1,0);

        for(int i = 0; i < m;i++){
            
            char s; 
            int l,r;cin>>s >> l >> r;
            
            if(l <= arr[i] && r >= arr[i]){
                if(s == '+'){
                    diff[i]++;
                }
                else{
                    diff[i]--;
                }
            }
            int current = 0;

            int maxVal = INT_MIN;
 
            for(int i = 0; i < n; i++){
                current += diff[i];
                arr[i] += current;
                maxVal = max(arr[i],maxVal);
            }
        
            cout << maxVal << " ";
           
          
        }     
         
        cout << "\n";

       
    } // main loop
}